_posts/2021-02-04-behind-the-scenes-devise.markdown

```diff
@@ -0,0 +1,162 @@
+---
+layout: post
+title: "Behind The Scenes: Devise"
+date: 2021-02-04 14:00:00
+categories: ["learning", "devise"]
+author: arieljuod
+---
+
+**Devise** is a well known solution for authentication in **Rails** applications. It's full featured (not only adds authentication but also password recovery, email changing, session timeout, locking, ip tracking, and more) and can be expanded to add even more (like JWT authentication).
+
+In this post, I'll go over the code related to the basic authentication process, how it relates to **Warden** and some of the magic behind it.
```

Comment:
What is Warden? Explain.

Reply:
There's a `What's Warden?` section later, do you think I should also add something here?